addons/fetchmail/models/fetchmail.py

```diff
@@ -108,22 +117,22 @@ def set_draft(self):
     def connect(self):
         self.ensure_one()
         if self.type == 'imap':
-            if self.is_ssl:
-                connection = IMAP4_SSL(self.server, int(self.port))
-            else:
-                connection = IMAP4(self.server, int(self.port))
+            # HACK https://bugs.python.org/issue38615
+            with patch.object(IMAP4, "_create_socket", side_effect=_create_socket_timeout):
```

Comment:
What if another thread on the same process is doing this in parallel at the same time? We really only want to monkey-patch the method once and for all when the module is loaded, that's it.